File: frontend/src/app/shared/models/schedule/reporte-dia-response.ts
import { TipoSesionEstado, TramoDiaResponse } from './detalle-dia-response';

/** Tiempo muerto detectado: hueco entre dos tramos de presencia (desconexión y retorno). */
export interface PresenciaGapResponse {
  inicio: string;
  fin: string;
  minutos: number | null;
  motivo: string | null;
  estadoAlDesconectar: string | null;
}

/** Una sesión de sub-estado del día (servicios / pausa activa / capacitación). */
export interface SesionEstadoResponse {
  tipo: TipoSesionEstado;
  inicio: string;
  fin: string | null;
  minutos: number | null;
  creadoPor: number | null;
}

/**
 * Reporte por empleado/día (admin/RRHH): desglose por tramo (re-derivado, cualquier fecha) + sesiones
 * de estado + tiempos muertos de presencia + totales. Sirve al detalle de asistencia del drawer.
 */
export interface ReporteDiaResponse {
  idEmpleado: number;
  fecha: string;
  tieneHorario: boolean | null;
  jornadaCerrada: boolean | null;

  tramos: TramoDiaResponse[] | null;
  sesiones: SesionEstadoResponse[] | null;
  tiemposMuertos: PresenciaGapResponse[] | null;

  almuerzoRealInicio: string | null;
  almuerzoRealFin: string | null;
  minutosAlmuerzoTomados: number | null;

  minutosObjetivoDia: number | null;
  minutosTrabajados: number | null;
  minutosBalance: number | null;
  minutosExtra: number | null;
  minutosCompensados: number | null;
}

File: schedule-service/src/main/java/pe/albrugroup/schedule_service/entity/response/asistencia/ReporteDiaResponse.java
package pe.albrugroup.schedule_service.entity.response.asistencia;

import lombok.Builder;
import lombok.Getter;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

/**
 * Reporte por empleado/dia (admin/RRHH): consolida el desglose por tramo (base + extras/compensables,
 * re-derivado, sin snapshot), las sesiones de sub-estado, los tiempos muertos de presencia y los totales.
 * Reusa la misma proyeccion que el read model de marcacion, para cualquier fecha.
 */
@Getter
@Builder
public class ReporteDiaResponse {

    private Long idEmpleado;
    private LocalDate fecha;
    private Boolean tieneHorario;
    private Boolean jornadaCerrada;

    private List<TramoDiaResponse> tramos;
    private List<SesionEstadoResponse> sesiones;
    private List<PresenciaGapResponse> tiemposMuertos;

    // Almuerzo real del dia.
    private LocalDateTime almuerzoRealInicio;
    private LocalDateTime almuerzoRealFin;
    private Integer minutosAlmuerzoTomados;

    // Totales del dia.
    private Integer minutosObjetivoDia;
    private Integer minutosTrabajados;
    private Integer minutosBalance;
    private Integer minutosExtra;
    private Integer minutosCompensados;
}
